go/config/config.go

```diff
@@ -15,101 +15,120 @@
 package config
 
 import (
+	"bytes"
+	"encoding/json"
 	"errors"
+	"fmt"
 	"io/ioutil"
-	"log"
 
-	fernet "github.com/fernet/fernet-go"
-	"gopkg.in/yaml.v2"
+	"github.com/spf13/viper"
 )
 
 // file is the Grafeas configuration file.
 type file struct {
-	Grafeas *GrafeasConfig `yaml:"grafeas"`
+	Grafeas GrafeasConfig `mapstructure:"grafeas"`
+}
+
+type GrafeasConfig struct {
+	API           *ServerConfig `mapstructure:"api"`
+	StorageType   string        `mapstructure:"storage_type"` // Natively supported storage types are "memstore" and "embedded"
+	StorageConfig *interface{}
```

Comment:
Also for this you might want to define a type so we know what it is when passing this around. I.e.,
type StorageConfig interface{}

Reply:
Amended to define a type.